### PR TITLE
Replace --xla_gpu_disable_autotune option with --xla_gpu_autotune_level

### PR DESCRIPTION
**Pull request by bas-aarts**
XLA performs autotuning for convolutions and GEMMs. for each algorithm
it runs some checks to check for out of bounds access or functional errors.
The latter in particular can take a lot of time, increasing compile time
by orders of magnitude. This hurts end-to-end execution time.

--xla_gpu_autotune_level enables clients to set a level.
        0: don't autotune
        1: autotune with uninitialized data
        2: autotune with initialized data
        3: autotune with initialized data, and reinitialize for inplace case
        4: autotune with initialized data, reinitialize, and check
The deafult is 4, not changing the current behaviour.

Change some tests accordingly.

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc [hunk 1]
```diff
@@ -320,14 +320,18 @@ GpuConvAlgorithmPicker::PickBestAlgorithmNoCacheCuda(
   const Shape& result_shape = instr->shape().tuple_shapes(0);
   int64 rng_state = 0;
 
-  const auto initialize_buffer = [&stream, &rng_state](
+  const HloModuleConfig& hlo_module_config = instr->GetModule()->config();
+  int32 conv_level = hlo_module_config.debug_options().xla_gpu_autotune_level();
```
Comment (timshen91) [inline tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc]: `autotune_level`

Comment (timshen91) [inline tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc]: Please add const to these variables.

Comment (bas-aarts) [reply tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc]: done

Comment (bas-aarts) [reply tensorflow/compiler/xla/service/gpu/gpu_conv_algorithm_picker.cc]: done

**Comment by bas-aarts** [comment]
Can someone review this PR please?

**Review by timshen91 — APPROVED** [review]

**Comment by timshen91** [comment]
> Can someone review this PR please?

Done. Sorry for the delay!

**Comment by bas-aarts** [comment]
Is there anything I can do on y end to resolve the unsuccessful check?

**Pull request merged**